packer/ansible/roles/remoterepos/tasks/main.yml

```diff
@@ -1,46 +1,152 @@
+
+
+- name: Populate dictionary with found repo info
+  set_fact:
+    repo_branch_organization_dictionary: "{{ repo_branch_organization_dictionary | default([]) + [{'repo': item.item, 'organization': organization | default('rackhd'), 'branch': item.stdout }] }}"
+  with_items: "{{ repos_found_in_organization_with_branch_stdout.results }}"
+- debug:
+    msg: "{{repo_branch_organization_dictionary}}"
+
+
+# For repos that were NOT forked into the organization
+# Also determine the branch exists in the organization specified, fallback to master
+
+- name: Build list of repos that were NOT found/forked in the {{ organization | default('rackhd') }} organization
+  set_fact:
+    repos_not_found_in_organization: "{{ r.results | selectattr('rc','equalto',128) | map(attribute='item') | list }}"  
+- debug:
+    msg: "{{ repos_not_found_in_organization }}"
+
+- name: Determine if branch from variable exists for the repos in the organization
+  shell: GIT_ASKPASS=true git ls-remote http://github.com/{{ organization | default('rackhd') }}//{{ item }} {{ branch | default('master') }} | wc -l | awk "/1/{print \"{{ branch | default('master') }}\"}/0/{print \"master\"}"
+  with_items: "{{ repos_not_found_in_organization }}"
+  register: "repos_not_found_in_organization_with_branch_stdout"
+  ignore_errors: true
+  changed_when: false
+- debug:
+    msg: "{{ repos_not_found_in_organization_with_branch_stdout }}"
+ 
+
+- name: Make a dictionary of branch names and repos to use from repos not found in organization
+  set_fact:
+    repo_branch_organization_dictionary: "{{ repo_branch_organization_dictionary | default([]) + [{'repo': item.item, 'organization': 'rackhd', 'branch': item.stdout }] }}"
+  with_items: "{{ repos_not_found_in_organization_with_branch_stdout.results }}"
+- debug:
+    msg: "{{ repo_branch_organization_dictionary }}"
+
+
+# Use the dictionary to clone down the repos
+- name: Clone down the dependent repos using dictionary data
+  git: repo=https://github.com/{{ item.organization }}/{{ item.repo }}
+       dest="{{ ansible_env.HOME }}/src/{{ item.repo }}"
+       accept_hostkey=true
+       version="{{item.branch}}"
+  with_items: "{{ repo_branch_organization_dictionary }}"
+
+
+# Continue with the NPM installation of the repos
+
 - name: install dev libraries for NPM installs
   apt: pkg={{ item }} state=installed
   with_items:
     - libkrb5-dev
     - unzip
   sudo: yes
 
-- name: Get RackHD from source
-  git: repo=https://github.com/rackhd/rackhd
-       dest="{{ ansible_env.HOME }}/src"
-       accept_hostkey=true
-       version="{{ branch | default('master') }}"
-
-- name: Update git repos with latest
-  shell: git fetch --all --prune
-  args:
-    chdir: "{{ ansible_env.HOME }}/src/{{ item }}"
-  with_items:
-    - on-http
-    - on-tftp
-    - on-dhcp-proxy
-    - on-taskgraph
-    - on-syslog
-    - on-wss
-    - on-tools
-    - ucs-service
-
-- name: Reset to the branch specified
-  shell: git reset --hard {{ branch }}
```

Comment:
I noticed this step is gone. although a brand new clone will ensure local HEAD aligns master HEAD, so I think this step is not necessary.. just raise this up in case you delete by accident.